File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    let answer = 0;
    
    function DFS(idx, sum){
        if(idx === numbers.length){
            if(sum === target)
                answer++;
            
            return;
        }
        
        DFS(idx+1, sum+numbers[idx]);
        DFS(idx+1, sum-numbers[idx]);
    }
    
    DFS(0, 0);
    
    return answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void DFS(int idx, int sum, vector<int> numbers, int target){
    if(idx == numbers.size()){
        if(sum == target)
            answer++;
        
        return;
    }
    
    DFS(idx+1, sum-numbers[idx], numbers, target);
    DFS(idx+1, sum+numbers[idx], numbers, target);
}

int solution(vector<int> numbers, int target) {
    DFS(0, 0, numbers, target);
    return answer;
}